src/main/scala/com/mwz/sonar/scala/scalastyle/ScalastyleSensor.scala

```diff
@@ -0,0 +1,186 @@
+      case styleError: StyleError[_] =>
+        log.debug(s"Processing ${styleError.clazz} for file ${styleError.fileSpec}.")
+
+        // Look up an active rule from the Scalastyle style error.
+        val rule = ScalastyleSensor.ruleFromStyleError(styleError)
+        rule.foreach { rule =>
+          ScalastyleSensor.openIssue(ScalastyleInspections.AllInspectionsByClass, styleError, rule)
+        }
+
+        // Log a warning if for some reason the rule was not found.
+        if (rule.isEmpty)
+          log.warn(
+            s"Scalastyle rule with key ${styleError.key} was not found" +
+            s"in the ${qualityProfile.getName} quality profile."
+          )
+      case e: StyleException[_] =>
+        log.error(s"Scalastyle exception (checker: ${e.clazz}, file: ${e.fileSpec.name}): ${e.message}.")
+      case _ =>
+        ()
+    }
+  }
+}
+
+private[scalastyle] object ScalastyleSensor {
+  final val SensorName = "Scalastyle Sensor"
+
+  /**
+   * Convert an active SonarQube rule to Scalastyle checker configuration.
+   */
+  def ruleToConfigurationChecker(rule: ActiveRule): Option[ConfigurationChecker] = {
+    val params = rule.params.asScala.map { case (k, v) => k -> v.trim }.toMap
+    val className: Option[String] = params.get(ScalastyleRulesRepository.RuleClassParam).filter(_.nonEmpty)
+    className.map { className =>
+      ConfigurationChecker(
+        className,
+        ScalastyleRulesRepository.severityToLevel(Severity.valueOf(rule.severity)),
+        enabled = true,
+        params,
+        customMessage = None,
+        customId = Some(rule.ruleKey.rule)
+      )
+    }
+  }
+
+  /**
+   * Get a list of files for analysis.
+   */
+  def fileSpecs(implicit context: SensorContext): Seq[FileSpec] = {
+    val predicates: FilePredicates = context.fileSystem.predicates
+    val files: Iterable[File] = context.fileSystem
+      .inputFiles(
+        predicates.and(
+          predicates.hasLanguage(Scala.LanguageKey),
+          predicates.hasType(InputFile.Type.MAIN)
+        )
+      )
+      .asScala
+      .map(f => new File(f.uri)) // Avoiding here to use InputFile.file, which is deprecated.
+
+    Directory.getFiles(Some(context.fileSystem.encoding.name), files)
+  }
+
+  /**
+   *  Look up an active rule from the Scalastyle style error.
+   */
+  def ruleFromStyleError(styleError: StyleError[FileSpec])(
+    implicit context: SensorContext
+  ): Option[ActiveRule] =
+    Option(
+      context
+        .activeRules()
+        .findByInternalKey(ScalastyleRulesRepository.RepositoryKey, styleError.key)
+    )
+
+  /**
+   * Open and new SonarQube issue for the given style error.
+   */
+  def openIssue(
+    inspections: Map[String, ScalastyleInspection],
+    styleError: StyleError[FileSpec],
+    rule: ActiveRule
+  )(implicit context: SensorContext): Unit = {
+    val predicates = context.fileSystem.predicates
+    val file: InputFile = context.fileSystem.inputFile(predicates.hasPath(styleError.fileSpec.name))
```

Comment:
We need to tests this with the example projects to make sure it works...
I'm already tired of file not found errors 😞